### DISCUSSION TITLE
How to silence warnings from and get status of an LSP server session?

### DISCUSSION THREAD

**Discussion by harry-xm** [redhat-developer/lsp4ij#493] [Q&A]
Open a Rust file in https://github.com/eunomia-bpf/eunomia-bpf/tree/master/ecli on macOS and LSP4IJ turns every warning message from `rust-analyzer` into a notification that is persisted in the Notifications panel.

![lsp4ij screenshot](https://github.com/user-attachments/assets/f82ccebc-4470-44b1-b041-189b5599f389)

![lsp4ij screenshot 2](https://github.com/user-attachments/assets/22e67dd4-41a2-4449-8544-4a5c0fbf8928)

Is it possible to suppress these notifications, and like the VSCode plugin, display a warning in the status bar? The user can then choose to open the LSP log buffer by clicking it. IMHO, that would be more user-friendly than a bunch of notifications.

![vscode screenshot](https://github.com/user-attachments/assets/bc8582b1-64d7-4e7a-9419-2041c4b90de6)

**Comment by angelozerr** [comment]
> Is it possible to suppress these notifications, and like the VSCode plugin, display a warning in the status bar? 

No, please create an issue to customize that with LSP API.

**Comment by angelozerr** [comment]
I close this disccusion, because I have created the issue https://github.com/redhat-developer/lsp4ij/issues/552
